apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/resources/claim_samhsa_methods.yaml

```diff
@@ -0,0 +1,706 @@
+- claimClass: CarrierClaim
```

Comment:
Clever solution! My only concern is that we're giving up a lot of compile-time safety by invoking all of this dynamically. Alternatively, we could have an adapter class for each claim type - `HHASamhsaAdapter`, `InpatientSamhsaAdapter`, etc. that would wrap each claim object and implement some interface or wrap some base class that we could operate on using common methods just like what you've done in these yaml files. It would be a bit more boilerplate than using reflection, but I think it would be a bit less prone to accidental breakage. Thoughts?

Reply:
If your concern about compile-time safety is that the methods may not exist, then the integration tests should alleviate that -- Every method in the yaml file is exercised. Or did you have a different concern?

The adapter solution would work, but I think it would be a bit more tightly coupled than the yaml solution. What I mean is that if, in the future, we add new SAMHSA fields, we would need only add two lines into the YAML file (column and method).

Reply:
If we leave this code as is and never change it, then yes, the tests will catch it. My main concern with these kinds of things is that it's more difficult to refactor and extend safely in the future. I think the concept here makes sense - with the adapter pattern we can still use `getFields()` as you have here and we should only need to add 1-2 lines of code to add the new field there.

Reply:
Been thinking about this, and I've come to agree with what you have been trying to get at: passing the methods along from the adapter with a `Supplier` object may be a better solution for the sake of maintainability Reflection would make the code a lot more compact, but:
1. Too easy to make a mistake, and
2. I'm not convinced that the execution time of invoking a method with reflection is advisable due to the volume of claims that we process at a time.